src/MongoDB.Driver/Core/Configuration/CryptClientSettings.cs

```
@@ -39,6 +39,11 @@ public sealed class CryptClientSettings
         /// </summary>
         public string CryptSharedLibSearchPath { get; }
 
+        /// <summary>
+        /// Gets the value of the expiration time for the DEK cache in ms.
+        /// </summary>
+        public long? DekCacheLifetimeMs { get; }
```

Is there a compelling reason to use a long here instead of a TimeSpan? .NET typically prefers TimeSpans for durations as they're easier to configure in terms of code and you don't need suffixes then like "Ms" on the end.

---

I think the issue here would be that `TimeSpan` has a greater granularity (microseconds) than the corresponding input for libmongocrypt (milliseconds).  
We could also approximate the value if microseconds are used, but I'm not super convinced. I agree that maybe it would be easier to read with `TimeSpan`, even though more verbose.

---

We could, or we could also throw an exception if there is microseconds.